clippy_lints/src/mixed_read_write_in_expression.rs

```diff
@@ -136,6 +137,15 @@ impl<'a, 'tcx> DivergenceVisitor<'a, 'tcx> {
 impl<'a, 'tcx> Visitor<'tcx> for DivergenceVisitor<'a, 'tcx> {
     fn visit_expr(&mut self, e: &'tcx Expr<'_>) {
         match e.kind {
+            // fix #10776
+            ExprKind::Block(block, ..) => {
+                if let Some(stmt) = block.stmts.first() && block.stmts.len() == 1 {
```

Comment:
The final expression should also be handled. e.g. `{ return }`

Reply:
thanks! fixed, didn't notice that before since rustfmt would always add a semicolon.

i've decided to make it return after checking the final expression since the original issue was that `return;` and `{ return; }` would be treated differently, yet be the same operation, so something like `{ _ = 1; return }` feels a bit silly.